csrc/runtime/executor_dispatch.h

```diff
@@ -0,0 +1,57 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#pragma once
+
+#include <runtime/executor.h>
+#include <runtime/executor_abstract.h>
+
+namespace nvfuser {
+
+// Simple stateless dispatch system for KernelExecutor, HostIrExecutor, and
+// ExprEvalExecutor
+class ExecutorDispatch {
+ public:
+  // Iterates through executors in priority order creating the first executor
+  // that returns true when checking their "supported" method
+  static std::unique_ptr<ExecutorAbstract> makeExecutor(
+      Fusion* fusion,
+      int64_t fusion_id = -1,
+      int64_t concrete_id = -1,
+      int64_t runtime_id = -1,
+      int64_t group_id = -1);
+
+  static void compile(
+      std::unique_ptr<ExecutorAbstract>& executor,
+      Fusion* fusion);
+
+  static void compile(
+      std::unique_ptr<ExecutorAbstract>& executor,
+      Fusion* fusion,
+      const KernelArgumentHolder& args,
+      const LaunchParams& launch_constraints,
+      CompileParams compile_params,
+      SchedulerType scheduler_type = SchedulerType::None);
+
+  static bool isCompiled(const std::unique_ptr<ExecutorAbstract>& executor);
+
+  static std::vector<at::Tensor> run(
+      std::unique_ptr<ExecutorAbstract>& executor,
+      KernelArgumentHolder& args,
+      std::vector<at::Tensor> outputs = {});
+
+  static std::vector<at::Tensor> run(
+      std::unique_ptr<ExecutorAbstract>& executor,
+      KernelArgumentHolder& args,
+      const LaunchParams& launch_constraints = LaunchParams(),
+      CompileParams compile_params = CompileParams(),
```

Comment:
I'm unsure if both are needed for compile and run. Conceptually, CompileParams should be used for compile only, and LaunchParams for run.

Reply:
Good point, will double check. This might have been blind copy-paste.

Reply:
done